packages/notifi-frontend-client/lib/client/NotifiFrontendClient.ts

```diff
@@ -251,6 +293,74 @@ export class NotifiFrontendClient {
     };
   }
 
+  private async logInWithWeb3(
+    signMessageParams: {
+      walletBlockchain: 'XION';
+      signMessage: XionSignMessageFunction;
+    },
+  ): Promise<Types.UserFragmentFragment> {
+    let user: Types.UserFragmentFragment | undefined = undefined;
+
+    if (this._configuration.walletBlockchain !== 'XION' || signMessageParams.walletBlockchain !== 'XION') {
+      throw new Error('Wallet blockchain must be XION for loginWithWeb3')
+    }
+    if (checkIsConfigWithDelegate(this._configuration)) {
+      const { delegatedAddress, delegatedPublicKey, delegatorAddress } = this._configuration;
+      const { nonce } = await this.beginLogInWithWeb3({
+        authAddress: delegatorAddress,
+        authType: 'COSMOS_AUTHZ_GRANT',
+      })
+
+      const message = `${SIGNING_MESSAGE}${nonce}}`;
+      const params = { walletBlockchain: 'XION', message, signMessage: signMessageParams.signMessage } as const
+      const signature = await this._signMessage({
+        signMessageParams: params,
+        timestamp: Math.round(Date.now() / 1000),
+      });
+
+      const { completeLogInWithWeb3 } = await this.completeLogInWithWeb3({
+        nonce,
+        signature,
+        signedMessage: message,
+        signingAddress: delegatedAddress,
+        signingPubkey: delegatedPublicKey
+      })
+      user = completeLogInWithWeb3.user
+
+    }
+    else if (checkIsConfigWithPublicKeyAndAddress(this._configuration)) {
```

Comment:
Do we suppose to have this `else if` case?
As far as I know from the code, it case will never be hit as long as the walletBlockchain is 'XION`. Not sure if I am wrong?